packages/cli/src/lib/container/docker_helper.ts

```diff
@@ -68,46 +69,120 @@ export const pullImage = function (resource: string) {
 
 export const getDockerfiles = function (rootdir: string, recursive: boolean) {
   const match = recursive ? './**/Dockerfile?(.)*' : 'Dockerfile*'
-  let dockerfiles: string[] = globSync(match, {
+  let dockerfiles = glob.sync(match, {
     cwd: rootdir,
     nodir: true,
   })
+
   if (recursive) {
-    dockerfiles = dockerfiles.filter((df: string) => df.match(/Dockerfile\.[\w]+$/))
+    dockerfiles = dockerfiles.filter(df => df.match(/Dockerfile\.[\w]+$/))
   } else {
-    dockerfiles = dockerfiles.filter((df: string) => df.match(/Dockerfile$/))
+    dockerfiles = dockerfiles.filter(df => df.match(/Dockerfile$/))
   }
 
-  return dockerfiles.map((file: string) => path.join(rootdir, file))
+  return dockerfiles.map(file => Path.join(rootdir, file))
+}
+
+export type dockerJob = {
+  name: string,
+  resource: string,
+  dockerfile: string,
+  postfix: number,
+  depth: number,
+}
+
+export type groupedDockerJobs = {
+  [processType: string]: dockerJob[],
 }
 
 export const getJobs = function (resourceRoot: string, dockerfiles: string[]) {
-  return dockerfiles
-  // convert all Dockerfiles into job Objects
-    .map(dockerfile => {
-      const match = dockerfile.match(DOCKERFILE_REGEX)
-      if (!match) throw new Error(`Invalid Dockerfile: ${dockerfile}`)
-      const proc: string = (match[1] || '.standard').slice(1)
-      const dockerfileJobs: Record<string, any> = {
+  const jobs: dockerJob[] = []
+  dockerfiles.forEach(dockerfile => {
+    const match = dockerfile.match(DOCKERFILE_REGEX)
+    if (match) {
+      const proc = (match[1] || '.standard').slice(1)
+      jobs.push({
         name: proc,
         resource: `${resourceRoot}/${proc}`,
         dockerfile: dockerfile,
-        postfix: path.basename(dockerfile) === 'Dockerfile' ? 0 : 1,
-        depth: path.normalize(dockerfile).split(path.sep).length,
-      }
-      return dockerfileJobs
-    })
+        postfix: Path.basename(dockerfile) === 'Dockerfile' ? 0 : 1,
+        depth: Path.normalize(dockerfile).split(Path.sep).length,
+      })
+    }
+  })
+
   // prefer closer Dockerfiles, then prefer Dockerfile over Dockerfile.web
-    .sort((a, b) => {
-      return a.depth - b.depth || a.postfix - b.postfix
-    })
+  jobs.sort((a, b) => {
+    return a.depth - b.depth || a.postfix - b.postfix
+  })
+
   // group all Dockerfiles for the same process type together
-    .reduce((jobs, job: Record<string, any>) => {
-      // if job.name === undefined) return
-      jobs[job.name] = jobs[job.name] || []
-      jobs[job.name].push(job)
-      return jobs
-    }, {})
+  const groupedJobs: groupedDockerJobs = {}
+  jobs.forEach(job => {
```

Comment:
Did we switch from `.reduce` to improve clarity here? Or was it not doing the right thing in a certain case?

Reply:
Just to improve clarity. There's an eslint warning to avoid usage of `Array.reduce` that suggested to change that into a `forEach` loop.